extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -1737,6 +1737,10 @@ export function shouldDebugCliServer() {
 }
 
 export class CliVersionConstraint {
+  // The oldest version of the CLI that we support. This is used to determine
+  // whether to show a warning about the CLI being too old on startup.
+  public static OLDEST_SUPPORTED_CLI_VERSION = new SemVer("2.7.6");
```

Comment:
Is it possible to get this value automatically from https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/supported_cli_versions.json, instead of having to manually update it? (No problem if that isn't feasible!)

Reply:
We're working on finalizing our deprecation strategy and it's probably going to be the case that we only need to support 1 year's worth of old CLIs. Given that we do a minor version bump every 3 months (coinciding with GHES releases), we only need to be supporting v2.10.x and later.

That being said, I don't think this is something that should be addressed in this PR. It's more of an FYI.

Once this is finalized, then I agree that we should have some processes around deprecations.